### PR TITLE
Content update - Attached events overview (user story 1878475) 

### PR DESCRIPTION
**Pull request by v-trisshores**
@adegeo this ported article and code project are ready for your review. [Internal preview](https://review.docs.microsoft.com/en-us/dotnet/desktop/wpf/events/attached-events-overview?view=netdesktop-5.0&branch=pr-en-us-1357)

The Framework version of this article incorrectly distinguishes an attached event from a regular routed event. For example:

- It states "The concept of an attached event enables you to add a handler for a particular event to an arbitrary element rather than to an element that actually defines or inherits the event." However, that definition is also applicable to non-attached routed events.
- It implies that the qualified event name syntax `<owner type>.<event name>` is unique to attached events. However, that syntax is also applicable to non-attached routed events.

An attached event is distinguished from a routed event by its backing code. Both this article and the Routed Events Overview article conflate an attached event with the XAML syntax that attaches a handler for a regular routed event (with CLR wrapper) to an arbitrary tree element (i.e. not the event source). Attaching a handler for a regular routed event to an arbitrary tree element doesn't transform the routed event into an attached event.

The clearest difference between attached and non-attached events is that an attached event can be defined in a non-element class (e.g. the `Mouse` class). This is because the attached event backing code calls the `UIElement.AddHandler` and `UIElement.RemoveHandler` methods on a passed in `UIElement` instance. In contrast, the CLR wrapper for non-attached routed events calls those methods directly on the owning class, which requires the owning class to derive from `UIElement`. Perhaps there are other differences I'm not seeing.

This [blog post](https://serialseb.com/blog/2007/01/25/attached-events-by-example-adding/), [stackoverflow discussion](https://stackoverflow.com/questions/6198184/wpf-attached-events-vs-non-attached-events#:~:text=Functional%20definition%3A%20A%20routed%20event,functionality%20as%20an%20attached%20event.), and my [tests](https://github.com/v-trisshores/wpf-test-projects/tree/main/AttachedEvents) helped clarify the attached event concept for me. Apart from reworking the introduction and inserting a few clarifications, the rest of the article has minimal rework.

### CODE REVIEW

File: dotnet-desktop-guide/net/wpf/events/attached-events-overview.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+When defining a custom attached event using the WPF model of basing attached events on [routed events](<xref:System.Windows.RoutedEvent>), use the <xref:System.Windows.UIElement.RaiseEvent%2A?displayProperty=nameWithType> method to raise an attached event on any <xref:System.Windows.UIElement> or <xref:System.Windows.ContentElement>. When raising a routed event, whether it's attached or not, you're required to designate an element in your element tree as the event source. That source is then reported as the `RaiseEvent` caller. For example, to raise the `AquariumFilter.Clean` attached routed event on `aquarium1`:
+
+:::code language="csharp" source="./snippets/attached-events-overview/csharp/MainWindow.xaml.cs" id="RaiseAttachedEvent":::
+
+:::code language="vb" source="./snippets/attached-events-overview/vb/MainWindow.xaml.vb" id="RaiseAttachedEvent":::
+
+Your code is responsible for determining which element is reported as the source in the element tree. In the preceding example, `aquarium1` is the event source.
```
Comment (adegeo) [inline dotnet-desktop-guide/net/wpf/events/attached-events-overview.md:91]: Line 91 seems to repeat what is at the end of line 85.

**Review by adegeo — CHANGES_REQUESTED** [review]
Minor thing to fix; looks amazing!

File: dotnet-desktop-guide/net/wpf/events/attached-events-overview.md (hunk 1 above)
Comment (v-trisshores) [reply dotnet-desktop-guide/net/wpf/events/attached-events-overview.md]: @adegeo Good catch. I've removed the redundant line portion of line 91.

**Review by adegeo — APPROVED** [review]

**Comment by v-trisshores** [comment]
@adegeo, thanks for the review!

**Pull request merged**